File: ruby/tabulation/howSum.rb
def howSum(targetSum, numbers)
    table = Array.new(targetSum+1, nil)
    table[0] = []

    (targetSum+1).times do |i|
        numbers.each do |n|
            if table[i] != nil
                table[i + n] = [*table[i], n]
            end
        end
    end

    table[targetSum]
end

p howSum(12, [31,4,6,7])

File: javascript/tabulation/howSum.js
const howSum = (targetSum, numbers) => {
    const table = Array(targetSum + 1).fill(null);
    table[0] = [];

    for (let i = 0; i<= targetSum; i++) {
        if (table[i] !== null) {
            for (let number of numbers) {
                table[i + number] = [...table[i], number]
            }
        }
    }
    return table[targetSum];
}

console.log(howSum(12, [35,5,7,4]));